docs: v10 (#828)
ansys/actions: doc/source/migrations/index.rst

  for information about detected issues and their remediation.

- **PR Documentation Deployment:** Added ``ansys/action/doc-deploy-pr`` to deploy HTML documentation
  for pull requests at ``https://<cname>/pull/<pr-number>/``. Documentation is automatically removed
  when the pull request is closed. Refer to :ref:`docs-deploy-pr-setup` for setup details.

- **Faster Package Installation:** Added a ``use-uv`` option (default: true) to ``build-library``,
  ``build-wheelhouse``, ``check-licenses``, ``check-vulnerabilities``, ``code-style``, ``doc-build``,
  ``doc-changelog``, ``doc-deply-changelog``, ``hk-package-clean-except``, ``hk-package-clean-untagged``,
  ``release-github``, and ``tests-pytest`` actions. This leverages `uv <https://docs.astral.sh/uv/>`_ for faster package installation,
  improving workflow speed for projects with multiple dependencies.

- **Dependency Groups Support:** ``doc-build`` and ``tests-pytest`` actions now support
  `PEP 735 <https://peps.python.org/pep-0735/>`_ dependency groups via the ``group-dependencies-name``
  input. Extras remain supported through the ``optional-dependencies-name`` input.

**Breaking Changes:**

> - **Python Version Support:** Dropped support for Python versions below ``3.9``.

Author: Correct, importlib was added since 3.8. But we made our check more restrictive because 3.8 is EOL (see [Roberto's suggestion](https://github.com/ansys/actions/pull/833#discussion_r2100416490)).
Thanks for clarifying the initial reference to pip-licenses.

  for information about detected issues and their remediation.

- **PR Documentation Deployment:** Added ``ansys/action/doc-deploy-pr`` to deploy HTML documentation
  for pull requests at ``https://<cname>/pull/<pr-number>/``. Documentation is automatically removed
  when the pull request is closed. Refer to :ref:`docs-deploy-pr-setup` for setup details.

- **Faster Package Installation:** Added a ``use-uv`` option (default: true) to ``build-library``,
  ``build-wheelhouse``, ``check-licenses``, ``check-vulnerabilities``, ``code-style``, ``doc-build``,
  ``doc-changelog``, ``doc-deply-changelog``, ``hk-package-clean-except``, ``hk-package-clean-untagged``,
  ``release-github``, and ``tests-pytest`` actions. This leverages `uv <https://docs.astral.sh/uv/>`_ for faster package installation,
  improving workflow speed for projects with multiple dependencies.

- **Dependency Groups Support:** ``doc-build`` and ``tests-pytest`` actions now support
  `PEP 735 <https://peps.python.org/pep-0735/>`_ dependency groups via the ``group-dependencies-name``
  input. Extras remain supported through the ``optional-dependencies-name`` input.

> 

Reviewer: As discussed offline there is a need for an extra bullet point on SBOM generation

**Breaking Changes:**

- **Python Version Support:** Dropped support for Python versions below ``3.9``.

- **JSON Builder Removal:** ``ansys/actions/doc-build`` no longer supports the ``JSON`` builder
  for documentation rendering.

- **Dependency Groups Parsing:** Replaced the ``toml`` library with ``tomli`` in ``doc-style``,
  ``doc-changelog``, and ``release-github`` actions due to improved support for ``pyproject.toml``
  files with nested dependency groups. The ``tomli-version`` input replaces the former
  ``toml-version`` input.

- **Build-Wheelhouse Changes:**  ``build-wheelhouse`` action no longer installs packages at the system level
  but uses a virtual environment instead. As this change will disrupt workflows relying on system-level
  installations for smoke tests immediately after calling the ``build-wheelhouse`` action, a new

>   ``smoke-test-module`` input has been added to enable on-the-fly smoke testing.

Reviewer 2: Thank you @moe-ad!

  ``doc-changelog``, and ``release-github`` actions due to improved support for ``pyproject.toml``
  files with nested dependency groups. The ``tomli-version`` input replaces the former
  ``toml-version`` input.

- **Build-Wheelhouse Changes:**  the ``build-wheelhouse`` action now installs packages in a virtual environment
  instead of at the system level. If you have subsequent workflow steps (e.g., running smoke tests) that depend on
  these packages, activate the virtual environment using the action's ``activate-venv`` output. For example:

  .. code-block:: yaml

    build-wheelhouse-and-smoke-test:
      name: Build wheelhouse and perform smoke test
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest]
        python-version: ['3.10', '3.11', '3.12', '3.13']
    steps:
      - name: Build wheelhouse
        id: build-wheelhouse
        uses: ansys/actions/build-wheelhouse@v10
        with:
          library-name: ${{ env.PACKAGE_NAME }}
          operating-system: ${{ matrix.os }}
          python-version: ${{ matrix.python-version }}
          check-licenses: true

>       - name: Perform smoke test

Reviewer 2: ```suggestion
      - name: Perform additional smoke tests
```

  :ref:`docs-deploy-pr-setup` for detailed information on setting up this action.

- A ``use-uv`` option that defaults to true has been added to the following actions: ``build-library``,
  ``build-wheelhouse``, ``check-licenses``, ``check-vulnerabilities``, ``code-style``, ``doc-build``,
  ``doc-changelog``, ``doc-deply-changelog``, ``hk-package-clean-except``, ``hk-package-clean-untagged``,
  ``release-github``, and ``tests-pytest``. With this option, these actions now take advantage of
  `uv <https://docs.astral.sh/uv/>`_ for fast package installation. This helps speed up workflows,
  especially for projects that require the install of multiple packages.

- `PEP 735 <https://peps.python.org/pep-0735/>`_ dependency groups are now automatically detected and used
  in:

  - the ``doc-build`` action where ``doc`` must be the group name.
  - the ``tests-pytest`` action where ``tests`` is the default group name targeted, or you may provide
    specific groups through the ``optional-dependencies-name`` input.

  .. warning::

    Extras (defined in ``[project.optional-dependencies]`` table in ``pyproject.toml``) are

>     not supported alongside dependency groups for the ``optional-dependencies-name`` input.

Reviewer: This will need to be updated once the `tests-pytest` action is updated as discussed yeterday

> Version ``v10.0``

Reviewer: ```suggestion
Version ``v10``
```
Let's use the major number only. This will help to standardize references to breaking changes, cf https://github.com/ansys/actions/issues/813

Reviewer 2: Agreed

  for common security and vulnerability issues, providing summary and detailed reports of detected issues.
  This action is based on `zizmor <https://docs.zizmor.sh/>`_ which is a static analysis tool for Github
  actions. Consult `zizmor audit rules <https://docs.zizmor.sh/audits/>`_ for more information about
  detected issues and how to remediate them.

- Added a new action named ``ansys/action/doc-deploy-pr``. This action deploys the HTML documentation for
  a pull request. The documentation is made available at ``https://<cname>/pull/<pr-number>/`` and
  the deployed documentation is removed automatically once the pull request is closed. Consult
  :ref:`docs-deploy-pr-setup` for detailed information on setting up this action.

- A ``use-uv`` option that defaults to true has been added to the following actions: ``build-library``,
  ``build-wheelhouse``, ``check-licenses``, ``check-vulnerabilities``, ``code-style``, ``doc-build``,
  ``doc-changelog``, ``doc-deply-changelog``, ``hk-package-clean-except``, ``hk-package-clean-untagged``,
  ``release-github``, and ``tests-pytest``. With this option, these actions now take advantage of

>   `uv <https://docs.astral.sh/uv/>`_ for fast package installation. This helps speed up workflows,
>   especially when bottlenecked by package installation.

Reviewer 3: ```suggestion
  `uv <https://docs.astral.sh/uv/>`_ for fast package installation. This helps speed up workflows,
  especially for projects that require the install of multiple packages.
```
